resen/Resen.py

```diff
@@ -662,13 +662,56 @@ def create_container(self,**input_kwargs):
         if image_id not in local_image_ids:
             print("Pulling image: %s" % image_name)
             print("   This may take some time...")
-            self.docker.images.pull(pull_image)
+            DockerHelper.stream_pull_image(pull_image)
+            image = self.docker.images.get(pull_image)
+            repo,digest = pull_image.split('@')
+            # When pulling from repodigest sha256 no tag is assigned. So:
+            image.tag(repo, tag=image_name)
             print("Done!")
 
         container_id = self.docker.containers.create(image_id,**create_kwargs)
 
         return container_id.id
 
+    @staticmethod
```

Comment:
Does this need to be a static method? No one should be using this outside of Resen, so I don't see the need for it to be static.

Reply:
Sure `stream_pull_image` doesn't need to be a static method. My reasoning was: This def doesn't really use any of the variables or methods of the class, it is independent. So it was a way of saying that. I am sure there are other ways, like having having the definition outside the class. But in favor of getting things moving I'll change that.